supporting-blog-content/agentic-rag/agent_rag_news_assistant.ipynb

```diff
@@ -0,0 +1,807 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "2b4b1e0f08209f34",
+   "metadata": {},
+   "source": [
+    "### Prerequisites:\n",
+    "\n",
+    "This AgenticRAG example uses GPT-4.1 via AzureChatOpenAI, which requires certain environment variables to be configured.\n",
+    "You can substitute any other LLM if preferred.\n",
+    "\n",
+    "Create a .env file in the same directory as your project and define the following variables:\n",
+    "- AZURE_OPENAI_ENDPOINT\n",
+    "- AZURE_OPENAI_KEY\n",
+    "- AZURE_OPENAI_DEPLOYMENT\n",
+    "- AZURE_OPENAI_API_VERSION\n",
+    "\n",
+    "You'll also need to set Elasticsearch environment variables as `ES_URL`, `ES_USERNAME` and `ES_PASSWORD` to set up connection with Elasticsearch vectorstore"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "id": "bce7dd63093dd6e0",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2025-10-29T13:50:06.920121Z",
+     "start_time": "2025-10-29T13:50:06.914925Z"
+    }
+   },
+   "source": [
+    "import os\n",
+    "import time\n",
+    "\n",
+    "from langchain_core.prompts import ChatPromptTemplate\n",
+    "from langchain_core.output_parsers import StrOutputParser\n",
+    "from langchain_core.runnables import RunnableSequence\n",
+    "from langchain_openai import AzureChatOpenAI\n",
+    "from langchain.chains import LLMChain, SequentialChain\n",
+    "from datasets import load_dataset\n",
+    "from langchain.schema import Document\n",
+    "from langchain_elasticsearch import ElasticsearchStore, SparseVectorStrategy\n",
+    "from dotenv import load_dotenv\n",
+    "from langchain_community.tools import DuckDuckGoSearchRun\n",
+    "from langgraph.graph import StateGraph, END, START\n",
+    "from typing import TypedDict, List, Literal\n",
+    "from loguru import logger\n",
+    "from pydantic import BaseModel, Field\n",
+    "from elasticsearch import Elasticsearch, NotFoundError, BadRequestError\n",
+    "from elasticsearch.exceptions import AuthenticationException\n",
+    "from IPython.display import Image\n",
+    "\n",
+    "load_dotenv()"
+   ],
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 23,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "execution_count": 23
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "f10b7273007fe477",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2025-10-27T16:07:42.075072Z",
+     "start_time": "2025-10-27T16:07:42.073032Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "#!pip3 install langchain langgraph langchain-openai langchain-elasticsearch langchain-community datasets python-dotenv loguru elasticsearch \"pydantic>=2.0\""
```

Comment:
This package installation should be above the imports, not below. Also uncommented